src/main/java/org/jabref/logic/integrity/YearContainedChecker.java

```diff
@@ -0,0 +1,29 @@
+package org.jabref.logic.integrity;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.strings.StringUtil;
+
+public class YearContainedChecker implements ValueChecker {
```

Comment:
On the one hand, shouldn't the year check simply check for 1950 to 2036 as number range?
On the other hand, this could be too loose when thinking of historical papers.

Therefore, OK for me as is.